### ISSUE TITLE
Original rancher.password reintroduced after upgrade.

### ISSUE THREAD

**Issue by kristoferlundgren** [rancher/os#2131]
**RancherOS Version: (ros os version)**
v1.1.0

**Where are you running RancherOS? (docker-machine, AWS, GCE, baremetal, etc.)**
VMware ESXi 5.5, installed to disk after booting from iso.

**Steps to reproduce:**

Boot from rancher.iso (I was using v1.1.0)
Install to disk using the following command:
```
sudo ros install -f -d /dev/sda --append "rancher.password=original"
```

ssh to rancheros as user "rancher", set new password to "newpass" using command:
(edit password in vi)
```
sudo ros config syslinux
sudo reboot
```

ssh to rancheros as "rancher", password "newpass".
-success

Upgrade to v1.1.0, (happens to be the same version as running version)
```
sudo ros os upgrade -f -i rancher/os:v1.1.0
```

System reboots with new rancheros.
ssh to rancheros as "rancher", password "newpass".
-fails!
ssh to rancheros as "rancher", password "original".
-success! (??)

View global.cfg:
```
sudo system-docker run --rm -it -v /:/host alpine cat /host/boot/global.cfg
```

File has original content.

**TL;DR**
The original rancher.password is kept somewhere, and re-introduced when upgrading.

**Comment by kristoferlundgren** [comment]
"rancher.password" is also reset to the original when upgrading from RancherOS v1.0.4 to v1.1.0 .
Since the "sudo ros config syslinux" command does not exist in v1.0.4 , you can instead edit the password using:
sudo system-docker run --rm -it -v /:/host alpine vi /host/boot/global.cfg

**Comment by niusmallnan** [comment]
Yeah, `rancher.password` is a kernel parameter, It will always be effective.
Recommend you to use ssh key.

Close it.
If you still have doubts, please reopen it and let me know.

**Comment by kristoferlundgren** [comment]
This is about the persistence of kernel parameters, not the effect of them. An upgrade effectively overwrites a reconfigured installation, not with defaults, but with parameters passed when originally installing to disk.

More details:
Parameters passed to the kernel is stored in host file /boot/global.cfg.
The parameter provided to the installation script is thus written to that file.
/boot/global.cfg is effectively overwritten when upgrading, with the content of the parameters passed on the commandline when originally installing.

I also tried adding "rancher.autologin=tty1" to /boot/global.cfg in a running system, a quite common secure parameter for management through console where a ssh key cannot be passed.
An upgrade removed the autologin configuration.

Issue still valid in v1.1.1 and even reproducible "upgrading" v1.1.1 -> v1.1.1

Please re-open (I cannot).

**Comment by boistordu** [comment]
I encountered the same thing now. from 1.5.2 to 1.5.3

I've installed rancheros or maybe I've activated a password a long time ago because ssh keys was not practical in my network setup. Then a few upgrades later I've changed this password to something stronger by editing kernel parameters as described there: [doc](https://rancher.com/docs/os/v1.x/en/installation/configuration/adding-kernel-parameters/#user-password)

and now after each upgrade I need to refresh the password. So where is this data stored to be restored at each upgrade? 
there is not trace of it in` /var/lib/rancher/conf/cloud-config.yml `nor `/var/lib/rancher/conf/cloud-config.d/*.*
`
